pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,664 @@
+			return err
+		},
+		createClusterRoleBinding: func(ctx context.Context, workspace logicalcluster.Name, crb *rbacv1.ClusterRoleBinding) error {
+			_, err := kubeClusterClient.Cluster(workspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+			return err
+		},
+		createClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, cw *tenancyv1alpha1.ClusterWorkspace) error {
+			_, err := kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Create(ctx, cw, metav1.CreateOptions{})
+			return err
+		},
+		getClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			return kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Get(ctx, name, metav1.GetOptions{})
+		},
+	}
+}
+
+type localInformersAccess struct {
+	getClusterWorkspace      func(logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error)
+	searchClusterRole        func(lcluster logicalcluster.Name, name string) (bool, error)
+	searchClusterRoleBinding func(lcluster logicalcluster.Name, name string) (bool, error)
+	synced                   func() bool
+}
+
+func buildLocalInformersAccess(kubeSharedInformerFactory coreexternalversions.SharedInformerFactory, kcpSharedInformerFactory kcpexternalversions.SharedInformerFactory) localInformersAccess {
+	clusterWorkspaceInformer := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Informer()
+	crInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Informer()
+	crbInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Informer()
+	clusterWorkspaceLister := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Lister()
+	crLister := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Lister()
+	crbLister := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Lister()
+
+	return localInformersAccess{
+		getClusterWorkspace: func(logicalCluster logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			parentLogicalCluster, workspaceName := logicalCluster.Split()
+			result, err := clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentLogicalCluster, workspaceName))
+			if err != nil {
+				return nil, err
+			}
+			return result, nil
+		},
+		searchClusterRole: func(workspace logicalcluster.Name, name string) (bool, error) {
+			_, err := crLister.Get(clusters.ToClusterAwareKey(workspace, name))
+			if kerrors.IsNotFound(err) {
+				return false, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		},
+		searchClusterRoleBinding: func(workspace logicalcluster.Name, name string) (bool, error) {
+			_, err := crbLister.Get(clusters.ToClusterAwareKey(workspace, name))
+			if kerrors.IsNotFound(err) {
+				return false, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		},
+		synced: func() bool {
```

Comment:
So in the case that we misconfigure the informers and this never syncs, how can an admin tell?

Reply:
Alternatively - can we add a `WaitForNamedCacheSync` to server startup for these somehow?

Reply:
`WaitForNamedCacheSync` is already there. This check here is because the server serves before all informers are up. It's the same behaviour many admission plugins and also authz have.